frontend_server/server.js

```diff
@@ -105,12 +106,30 @@ app.use('/api', apiProxy)
 
 app.use(compression())
 
+app.get('/rate-limit-status', (req, res) => {
+  res.json({
+    rateLimitReached: RATE_LIMIT_REACHED,
+  })
+})
+
+const aiInsightCache = new Map()
+const CACHE_TTL_MS = 24 * 60 * 60 * 1000
```

Comment:
this is good for now; we should discuss with @kccrtv  and maybe @alinix1 and @JDemlow as well and see what would feel best for a user experience here. maybe it could be longer than 24 hours if we end up adding some other UI for insight stuff like regenerating a response, or user giving a thumbs up/down to the AI response, etc